### PR TITLE
#2623 #2587 Investment visual and invest whole balance fixes

### PR DESCRIPTION
**Pull request by drakkein**

### CODE REVIEW

File: app/components/modals/Modal.module.scss [hunk 1]
```diff
@@ -1,6 +1,8 @@
 .modal {
   width: 100%;
   padding-top: 12px;
+  overflow-x: hidden;
+
```
Comment (desfero) [inline app/components/modals/Modal.module.scss]: two empty spaces not needed here, maybe we can enforce this somehow on sasslint level?

File: app/components/modals/Modal.module.scss [hunk 2]
```diff
@@ -1,6 +1,8 @@
 .modal {
   width: 100%;
   padding-top: 12px;
+  overflow-x: hidden;
```
Comment (desfero) [inline app/components/modals/Modal.module.scss]: btw, what's the exact reason for `overflow: hidden`?

File: app/components/modals/tx-sender/investment-flow/Investment.tsx [hunk 3]
```diff
@@ -166,81 +165,57 @@ export const InvestmentSelectionComponent: React.FunctionComponent<IProps> = ({
...
-                    decimalLimit: 4,
-                    integerLimit: 15, // integer limit due to weird behavior on large inputs
-                  })}
-                  onChange={e => changeEthValue(e.target.value)}
-                />
-              )}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) => changeEthValue(e.target.value)}
             />
-            <a
+            <Button
```
Comment (desfero) [inline app/components/modals/tx-sender/investment-flow/Investment.tsx]: 👍

File: app/components/shared/forms/fields/FormMaskedInput.unsafe.tsx [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { isNonValid } from "./utils.unsafe";
 export interface IMaskedFormInputExternalProps extends IFormInputRawExternalProps {
   mask: maskArray;
   guided?: boolean;
+  onChange?: Function;
```
Comment (desfero) [inline app/components/shared/forms/fields/FormMaskedInput.unsafe.tsx]: hm, i would prefer to use `() => void` for consistency

File: app/modules/tx/transactions/investment/utils.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import { EInvestmentType } from "../../../investment-flow/reducer";
+import {
+  selectLiquidEtherBalance,
+  selectLiquidEuroTokenBalance,
+  selectLockedEtherBalance,
+  selectLockedEuroTokenBalance,
+} from "../../../wallet/selectors";
+import { selectTxGasCostEthUlps } from "../../sender/selectors";
+
+export const getWalletBalance = (state: IAppState): string => {
```
Comment (desfero) [inline app/modules/tx/transactions/investment/utils.ts]: imo, for me this is a selector as you pass whole app state

File: app/modules/tx/transactions/investment/utils.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      return true;
+    case EInvestmentType.NEur:
+    case EInvestmentType.ICBMnEuro:
+      return false;
+    default:
+      throw Error("Incorrect investment type");
+  }
+};
+
+export const getMaximumInvestment = (state: IAppState): string => {
```
Comment (desfero) [inline app/modules/tx/transactions/investment/utils.ts]: imo, for me this is a selector not an util as you pass whole app state

File: app/components/modals/Modal.module.scss (hunk 2 above)
Comment (drakkein) [reply app/components/modals/Modal.module.scss]: with `overflow: auto` we will have nice scroll, so let's leave it this way

File: app/components/modals/Modal.module.scss (hunk 1 above)
Comment (drakkein) [reply app/components/modals/Modal.module.scss]: there is no such ruleset available :(

File: app/components/shared/forms/fields/FormMaskedInput.unsafe.tsx (hunk 4 above)
Comment (drakkein) [reply app/components/shared/forms/fields/FormMaskedInput.unsafe.tsx]: ok :+1:

**Comment by Moejoe90** [comment]
This can catch up with production merge pt2

**Pull request merged**